src/internal/module/types.go

```diff
@@ -0,0 +1,22 @@
+package module
+
+import "fmt"
+
+type Version struct {
+	Version string `json:"version"` // The version number of the provider. Correlates to a tag in the module repository
+}
+
+type MetadataFile struct {
+	Versions []Version `json:"versions"`
+}
+
+type Module struct {
+	Namespace    string // The module namespace
+	Name         string // The module name
+	TargetSystem string // The module target system
+}
+
+func (m *Module) RepositoryURL() string {
```

Comment:
**(nitpick)** I find that the receiver does not have to be pointer because the method is a "getter".

```suggestion
func (m Module) RepositoryURL() string {
```

Reply:
63df37a7bb917867ba51dc472234871237d550a7